### ISSUE TITLE
Fresh orchest-controller installation: Environment stuck on "PENDING"

### ISSUE THREAD

**Issue by glxhb** [orchest/orchest#991]
**Describe the bug**
Installed orchest on a fresh microk8s cluster using the orchest-cli tool, and assigned an IP address to the orchest-webserver service using metallb. 

I can access and navigate the dashboard just fine, but when I create a project, it gets stuck at creating the environment.

I have tried creating a fresh VM but without luck.

Hypervisor: ESXI 7.0.2U
OS: Ubuntu 22.04 Server LTS
RAM: 64GB
HDD: 248GB
Client Browser: Safari Version 15.5 (17613.2.7.1.8)
Orchest Version: v2022.05.3

**Screenshots**
![Screenshot 2022-05-26 at 10 44 56](https://user-images.githubusercontent.com/68229766/170452927-2c86be55-56f3-440e-a6e4-9a252204b2af.png)

**Comment by fruttasecca** [comment]
Hi @glxhb, thanks for reporting the issue. Could you:
- check out `Settings` -> `Customized JupyterLab`, and try to build?

I would also be interested in seeing the following logs when you try to build an environment:

- `kubectl logs -n orchest -f deployment/orchest-api`
- `kubectl logs -n orchest -f deployment/orchest-webserver`
- `kubectl logs -n orchest -f deployment/celery-worker`

**Comment by glxhb** [comment]
Configuring Jupiter lab gets stuck as well.

Here are the files. worker_builds-stdout and celery_builds.log did not exist

[celery-worker.txt](https://github.com/orchest/orchest/files/8777964/celery-worker.txt)
[orchest-api.txt](https://github.com/orchest/orchest/files/8777966/orchest-api.txt)
[stderr.txt](https://github.com/orchest/orchest/files/8777967/stderr.txt)
[webserver.txt](https://github.com/orchest/orchest/files/8777968/webserver.txt)

**Comment by fruttasecca** [comment]
Thanks for the logs, at a first glance it looks like a race condition that we are trying to solve in https://github.com/orchest/orchest/pull/966. Have you tried stopping and restarting Orchest already? On start it setup db connections and schema migrations again, that's where the race condition is happening.

**Comment by glxhb** [comment]
Can't believe I didn't try a restart. That's IT 101. Its building now. It might now be stuck at "storing image" though. Its been running for approx 15 mins now, I will give it a bit longer and then try your second suggested solution.

**Comment by fruttasecca** [comment]
>It might now be stuck at "storing image" though. 

That's strange, how big is the image?

**Comment by glxhb** [comment]
Its only the base image I'm trying to store.network, disk and cpus shouldn't be an issue. I have previously used the minikube version of orchest without problems. It does really seem stuck though. Could it be anything with the storage class?
<img width="1679" alt="Screenshot 2022-05-26 at 12 39 45" src="https://user-images.githubusercontent.com/68229766/170472234-912eee66-46c5-4494-ad12-1ee56c893fd3.png">

**Comment by fruttasecca** [comment]
I am indeed wondering if it could be the storage class (@nhaghighat ), anything interesting from
>kubectl logs -n orchest -f deployment/docker-registry
?

**Comment by glxhb** [comment]
The docker-registry logs:
[docker-registry.log](https://github.com/orchest/orchest/files/8778402/docker-registry.log)

There were two image-build-tasks, both of them having "Not Ready" 
![Screenshot 2022-05-26 at 13 02 02](https://user-images.githubusercontent.com/68229766/170476151-7b05e6e4-af66-48ef-a9e1-44bc49a2658c.png)


[build-task1.log](https://github.com/orchest/orchest/files/8778484/build-task1.log)
[build-task2.log](https://github.com/orchest/orchest/files/8778485/build-task2.log)

**Comment by fruttasecca** [comment]
Can you send me the logs of the `main` container of the build task? e.g. `kubectl logs -n orchest image-build-task-<uuid> main`

**Comment by glxhb** [comment]
Here you go

```
STEP 1/6: FROM docker.io/orchest/base-kernel-py:v2022.05.3
STEP 2/6: LABEL _orchest_project_uuid=8f5ac3ba-7b6c-47ea-b650-aceaeee39bba
--> Using cache 65e5bbd20177c5bb32808d7f012cf4655a46ad811260107d5d9e11b22d0225eb
--> 65e5bbd2017
STEP 3/6: LABEL _orchest_environment_uuid=978d669d-72d9-4393-9d66-86e27a1de4eb
--> Using cache aa51fc5fd575307c4e85c28b94ea91bde7f861dddf6a3573eab90d58bea676d9
--> aa51fc5fd57
STEP 4/6: WORKDIR /project-dir
--> Using cache acc8bb8c2ee31a5c5d0784f888e72bc1b19b6239fc42099d0a6e09b83714ffe1
--> acc8bb8c2ee
STEP 5/6: COPY . .
--> Using cache bfe0d5d95bedcad80b94843c475216a5e5629be34049ed723f2912ba0555b88a
--> bfe0d5d95be
STEP 6/6: RUN ((if [ $(id -u) = 0 ]; then chown -R :$(id -g) . > /dev/null 2>&1  && find . -type d -exec chmod g+rwxs {} \; > /dev/null 2>&1  && find . -type f -exec chmod g+rwx {} \; > /dev/null 2>&1  && chmod g+rwx . > /dev/null 2>&1 ; else sudo chown -R :$(id -g) . > /dev/null 2>&1  && sudo find . -type d -exec chmod g+rwxs {} \; > /dev/null 2>&1  && sudo find . -type f -exec chmod g+rwx {} \; > /dev/null 2>&1  && sudo chmod g+rwx . > /dev/null 2>&1 ; fi) || ! echo "The base image must have USER root or "sudo" must be installed, "find" must also be installed.") && bash < .orchest-reserved-env-setup-script-8f5ac3ba-7b6c-47ea-b650-aceaeee39bba-978d669d-72d9-4393-9d66-86e27a1de4eb.sh && echo _ORCHEST_RESERVED_LOG_FLAG_ && (if [ $(id -u) = 0 ]; then rm .orchest-reserved-env-setup-script-8f5ac3ba-7b6c-47ea-b650-aceaeee39bba-978d669d-72d9-4393-9d66-86e27a1de4eb.sh; else sudo rm .orchest-reserved-env-setup-script-8f5ac3ba-7b6c-47ea-b650-aceaeee39bba-978d669d-72d9-4393-9d66-86e27a1de4eb.sh; fi) || (echo _ORCHEST_RESERVED_ERROR_FLAG_ && PRODUCE_AN_ERROR)
--> Using cache 4a694b513dbf1e1238ee6787a593b2385fcd1090ca6b48926407c045f6517124
COMMIT docker-registry.orchest.svc.cluster.local/orchest-env-8f5ac3ba-7b6c-47ea-b650-aceaeee39bba-978d669d-72d9-4393-9d66-86e27a1de4eb:3
--> 4a694b513db
Successfully tagged docker-registry.orchest.svc.cluster.local/orchest-env-8f5ac3ba-7b6c-47ea-b650-aceaeee39bba-978d669d-72d9-4393-9d66-86e27a1de4eb:3
Successfully tagged docker-registry.orchest.svc.cluster.local/orchest-env-8f5ac3ba-7b6c-47ea-b650-aceaeee39bba-978d669d-72d9-4393-9d66-86e27a1de4eb:2
4a694b513dbf1e1238ee6787a593b2385fcd1090ca6b48926407c045f6517124
Getting image source signatures
Copying blob sha256:00651388389851619cc6ccf7fe56f0d5ed8fa404bdb6eb42a65e761a96ddb269
Copying blob sha256:3e2952fbf4510eee05257fce4c543769c4fb6d428ee799699de07b522decd00a
Copying blob sha256:71f99e5f538ee942395bb2c011fb19a79288219844d4ed35e5c5e3e80f9b1422
Copying blob sha256:c947c9bc308b735c36f51fabc1bb1ccbde09fc4dec51ea9a4d18cb3ced1bd403
Copying blob sha256:fb7ad75a23ef39503329e3f8f6a9991e8a5f4d8e5bdf158553ea79ba457c1170
Copying blob sha256:b653887bbbe7d38fda625a812f21b8c979d50a56cb5b8b4ab5b62e1f21d6c460
Copying blob sha256:21bfdb00f189260142c0433132c50ba112ad23137cfe7bab52d7188d6079e565
Copying blob sha256:d838485a2109d65cd53a647f73f06c6656cfc60ee41e66230320c4ea3cae17e9
Copying blob sha256:d6297af286a631c818c46865af741d654a5863642145a4e8e4be68d7d557fd05
Copying blob sha256:68a85fa9d77ecac87de23805c4be8766bda08a86787e324036cbcf84b62640fa
Copying blob sha256:b540354327258c792306a1849ff82318d1cf0d0e46d6cf8e1d6c83a7997bb67e
Copying blob sha256:8ce28733f84bb66bdb3f4f9200b730a3e0c62a9a47853a81ea79e4748aac04c7
Copying blob sha256:3c072d0153c87072ad120a0194231f68bfcc6a6ff977bdbf4419e5b995a6ec8a
Copying blob sha256:21192b89a18a46c50d77341b3fc20f1b8ef738fb7f4e3ce2a09c1f42f5738e86
Copying blob sha256:a568346ab0d7b771c8c9406bc82f989eaef82830dd21720010c71cb7b145c0ca
Copying blob sha256:f8d46cf235dbf6a64900f87484ddf97c7421d7a8e5ccf442cc26cfb026435f25
Copying blob sha256:b29c7a072e0f210b372f5f2cb51fe48e1dfa186fe0ffe372e5c59196ade0e127
Copying blob sha256:ac50136ce860aecd2324809c4e6a1f734fa84c88a25e10d36be2295e9b77c981
Copying blob sha256:942facd2b7c1c70b100c10d4e7a38c223ad678869f6038f782218205dce46603
Copying blob sha256:113b976230395e3f805320e8e98e2a95b31a1f121ac060bac3c5b06e7c8ba488
Copying blob sha256:85a9a01375dcfc793d0014ebd5a22a44d42bf0389b64d57d80e522666f776afb
Copying blob sha256:6e50fc54074bcab15a9f7665da89dd20ffdd0cce7406a036ab3cf09efdb630b3
Copying blob sha256:13d887bd5ae745659682d850e005fa2f6f1ebd978825fee6477d3b9cf0ec29b0
Copying blob sha256:c4c9fe9c51bc52753dcf6788a7cd95f63130a7ca158e7946a515041e4be9531d
Copying blob sha256:10915d8ab9c3dfca79e73c8c4fa5b7a2e6a1f3e4f247235aafc4263b77d526b8
Copying config sha256:4a694b513dbf1e1238ee6787a593b2385fcd1090ca6b48926407c045f6517124
Writing manifest to image destination
Storing signatures
```

**Comment by fruttasecca** [comment]
I am brainstorming a bit in the background, as a cheap experiment, mind deleting and recreating the project you are working with? I am wondering if some broken state was left by issue we solved with a restart. Judging by the logs it looks like the build has gone through, although I don't understand the "NotReady" state of the pods.

**Comment by fruttasecca** [comment]
Mind if I ask you to provide the logs mentioned here (https://github.com/orchest/orchest/issues/991#issuecomment-1138319256) again? Would like to see them after the env build gets stuck like that

**Comment by nhaghighat** [comment]
> I am indeed wondering if it could be the storage class (@nhaghighat ), anything interesting from

If it was storage class issue, then orchest or docker-registry weren't able to reach running state.

**Comment by fruttasecca** [comment]
@glxhb after installing Orchest on microk8s I was able to reproduce, I will try to fix this tomorrow or in the following days

**Comment by yannickperrenet** [comment]
@nhaghighat Once we merge #1048, can you write up (probably a "copy-paste" from the new docs) how to install on microk8s? :muscle:

**Comment by yannickperrenet** [comment]
With the merge of #1048, the installation docs are now include a detailed write-up of how to run Orchest on MicroK8s. Check them out here: https://docs.orchest.io/en/latest/getting_started/installation.html#install-orchest

The changes from the PR will be included in the next Orchest release (`v2022.07.2`) :rocket:
